java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/accessor/ArrowFlightJdbcAccessor.java

```diff
@@ -250,6 +251,57 @@ public Reader getNCharacterStream() throws SQLException {
     return !type.isPrimitive() && wasNull ? null : type.cast(value);
   }
 
+  /**
+   * Formats a period similar to Oracle INTERVAL YEAR TO MONTH data type<br>.
+   * For example, the string "+21-02" defines an interval of 21 years and 2 months.
+   */
+  public static String formatIntervalYear(final org.joda.time.Period p) {
+    long months = p.getYears() * (long) DateUtility.yearsToMonths + p.getMonths();
+    boolean neg = false;
+    if (months < 0) {
+      months = -months;
+      neg = true;
+    }
+    final int years = (int) (months / DateUtility.yearsToMonths);
+    months = months % DateUtility.yearsToMonths;
+
+    return String.format("%c%03d-%02d", neg ? '-' : '+', years, months);
+  }
+
+  /**
+   * Formats a period similar to Oracle INTERVAL DAY TO SECOND data type.<br>.
+   * For example, the string "-001 18:25:16.766" defines an interval of
+   * - 1 day 18 hours 25 minutes 16 seconds and 766 milliseconds.
+   */
+  public static String formatIntervalDay(final org.joda.time.Period p) {
+    long millis = p.getDays() * (long) DateUtility.daysToStandardMillis + millisFromPeriod(p);
+
+    boolean neg = false;
+    if (millis < 0) {
+      millis = -millis;
+      neg = true;
+    }
+
+    final int days = (int) (millis / DateUtility.daysToStandardMillis);
+    millis = millis % DateUtility.daysToStandardMillis;
+
+    final int hours = (int) (millis / DateUtility.hoursToMillis);
+    millis = millis % DateUtility.hoursToMillis;
+
+    final int minutes = (int) (millis / DateUtility.minutesToMillis);
+    millis = millis % DateUtility.minutesToMillis;
+
+    final int seconds = (int) (millis / DateUtility.secondsToMillis);
+    millis = millis % DateUtility.secondsToMillis;
+
+    return String.format("%c%03d %02d:%02d:%02d.%03d", neg ? '-' : '+', days, hours, minutes, seconds, millis);
+  }
+
+  public static int millisFromPeriod(org.joda.time.Period period) {
+    return period.getHours() * DateUtility.hoursToMillis + period.getMinutes() * DateUtility.minutesToMillis +
+        period.getSeconds() * DateUtility.secondsToMillis + period.getMillis();
+  }
+
```

Comment:
Can these methods be in an utility class? Let's avoid static methods that can potentially be used elsewhere inside the accessors.

Reply:
Done.